source/change-streams.rst

```diff
@@ -339,19 +421,54 @@ Single Server Topologies
 
 Presently, change streams cannot be initiated on single server topologies as they do not have an oplog.  Drivers MUST NOT throw an exception in this scenario, but instead rely on an error returned from the server.  This allows for the server to seamlessly introduce support for this in the future, without need to make changes in driver code.
 
+startAtClusterTime
+^^^^^^^^^^^^^^^^^^
+
+:since: 4.0
+
+``startAtClusterTime`` specifies that a ``changeStream`` will only return changes that occurred after the specified ``Timestamp``.
+
+The server expects ``startAtClusterTime`` as a bson document of the form
+
+.. code:: typescript
+
+    {
+      ts: Timestamp;
+    }
+
+Drivers MUST allow users to specify a ``startAtClusterTime`` option in the ``Collection.watch`` and ``Database.watch`` helpers. They SHOULD allow users to specify this value as a raw ``Timestamp``, and SHOULD NOT force users to wrap the timestamp as the server specifies.
+
+``startAtClusterTime`` and ``resumeAfter`` are mutually exclusive; if both ``startAtClusterTime`` and ``resumeAfter`` are set, the server will return an error. Drivers MUST NOT throw a custom error, and MUST defer to the server error.
+
+If neither ``startAtClusterTime`` or ``resumeAfter`` are specified, and the server version is >= ``4.0`` drivers MUST set a default ``startAtClusterTime`` value using a ``$clusterTime`` from any server response (ex: ``ismaster``). This allows change streams to be resumed before the first notification is received.
```

Comment:
> This allows change streams to be resumed before the first notification is received

I'm not sure what this means.  What does this allow an application to do that can't be done in 3.6?

Reply:
If I understand the spec correctly, the idea is that the following occurs:

1. An aggregation command is sent for a collection for which there are not currently changes.
2. The server replies with a new cursor, but without a resume token.
3. The driver sends a getMore to get changes.
4. A Resumable Error occurs.
5. A change `X` occurs in the collection
6. The changeStream closes the old cursor, and creates a new cursor. Since `startAtClusterTime` is specified to before `X` occurred, `X` will be returned with the new cursor.